<h1>DATA ANALYTICS PROJECT</h1>

<h2>United Kingdom Road Accident Data Analysis</h2>

<h3>Inclusive years: 2019 - 2022</h3>

In [1]:
pip install numpy

Defaulting to user installation because normal site-packages is not writeable
Note: you may need to restart the kernel to use updated packages.



[notice] A new release of pip is available: 24.3.1 -> 25.0.1
[notice] To update, run: C:\Users\LAB5-StudentPC05\AppData\Local\Microsoft\WindowsApps\PythonSoftwareFoundation.Python.3.13_qbz5n2kfra8p0\python.exe -m pip install --upgrade pip


In [2]:
pip install pandas

Defaulting to user installation because normal site-packages is not writeable
Note: you may need to restart the kernel to use updated packages.



[notice] A new release of pip is available: 24.3.1 -> 25.0.1
[notice] To update, run: C:\Users\LAB5-StudentPC05\AppData\Local\Microsoft\WindowsApps\PythonSoftwareFoundation.Python.3.13_qbz5n2kfra8p0\python.exe -m pip install --upgrade pip


In [3]:
pip install seaborn

Defaulting to user installation because normal site-packages is not writeable
Note: you may need to restart the kernel to use updated packages.



[notice] A new release of pip is available: 24.3.1 -> 25.0.1
[notice] To update, run: C:\Users\LAB5-StudentPC05\AppData\Local\Microsoft\WindowsApps\PythonSoftwareFoundation.Python.3.13_qbz5n2kfra8p0\python.exe -m pip install --upgrade pip


In [4]:
pip install matplotlib

Defaulting to user installation because normal site-packages is not writeable
Note: you may need to restart the kernel to use updated packages.



[notice] A new release of pip is available: 24.3.1 -> 25.0.1
[notice] To update, run: C:\Users\LAB5-StudentPC05\AppData\Local\Microsoft\WindowsApps\PythonSoftwareFoundation.Python.3.13_qbz5n2kfra8p0\python.exe -m pip install --upgrade pip


<h2>IMPORTING LIBRARIES</h2>

In [5]:
import numpy as np 
import pandas as pd 
import seaborn as sns 
import matplotlib.pyplot as plt
import warnings
warnings.filterwarnings('ignore')

In [6]:
accident = pd.read_csv('C:\\Users\\LAB5-StudentPC05\\Desktop\\ITD3A\\dataset_sioson\\accident_data.csv')

<h2>DATA PREPARATION</h2>

In [7]:
accident

Index Accident_Severity Accident Date   Latitude  \
0       200701BS64157           Serious    05/06/2019  51.506187   
1       200701BS65737           Serious    02/07/2019  51.495029   
2       200701BS66127           Serious    26/08/2019  51.517715   
3       200701BS66128           Serious    16/08/2019  51.495478   
4       200701BS66837            Slight    03/09/2019  51.488576   
...               ...               ...           ...        ...   
660674  201091NM01760            Slight    18/02/2022  57.374005   
660675  201091NM01881            Slight    21/02/2022  57.232273   
660676  201091NM01935            Slight    23/02/2022  57.585044   
660677  201091NM01964           Serious    23/02/2022  57.214898   
660678  201091NM02142           Serious    28/02/2022  57.575210   

                   Light_Conditions           District Area  Longitude  \
0             Darkness - lights lit  Kensington and Chelsea  -0.209082   
1                          Daylight  Kensington and Chelsea  -0.173647   
2       Darkness - lighting unknown  Kensington and Chelsea  -0.210215   
3                          Daylight  Kensington and Chelsea  -0.202731   
4             Darkness - lights lit  Kensington and Chelsea  -0.192487   
...                             ...                     ...        ...   
660674                     Daylight                Highland  -3.467828   
660675       Darkness - no lighting                Highland  -3.809281   
660676                     Daylight                Highland  -3.862727   
660677       Darkness - no lighting                Highland  -3.823997   
660678                     Daylight                Highland  -3.895673   

        Number_of_Casualties  Number_of_Vehicles Road_Surface_Conditions  \
0                          1                   2                     Dry   
1                          1                   2             Wet or damp   
2                          1                   3                     Dry   
3                          1                   4                     Dry   
4                          1                   2                     Dry   
...                      ...                 ...                     ...   
660674                     2                   1                     Dry   
660675                     1                   1            Frost or ice   
660676                     1                   3            Frost or ice   
660677                     1                   2             Wet or damp   
660678                     1                   1             Wet or damp   

                 Road_Type Urban_or_Rural_Area     Weather_Conditions  \
0       Single carriageway               Urban     Fine no high winds   
1       Single carriageway               Urban  Raining no high winds   
2                      NaN               Urban                    NaN   
3       Single carriageway               Urban     Fine no high winds   
4                      NaN               Urban                    NaN   
...                    ...                 ...                    ...   
660674  Single carriageway               Rural     Fine no high winds   
660675  Single carriageway               Rural     Fine no high winds   
660676  Single carriageway               Rural     Fine no high winds   
660677  Single carriageway               Rural     Fine no high winds   
660678    Dual carriageway               Rural  Snowing no high winds   

                                Vehicle_Type  
0                                        Car  
1                                        Car  
2                      Taxi/Private hire car  
3       Bus or coach (17 or more pass seats)  
4                              Other vehicle  
...                                      ...  
660674                                   Car  
660675                                   Car  
660676                                   Car  
660677                 Motorcycle over 500cc  
660

In [8]:
accident.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 660679 entries, 0 to 660678
Data columns (total 14 columns):
 #   Column                   Non-Null Count   Dtype  
---  ------                   --------------   -----  
 0   Index                    660679 non-null  object 
 1   Accident_Severity        660679 non-null  object 
 2   Accident Date            660679 non-null  object 
 3   Latitude                 660654 non-null  float64
 4   Light_Conditions         660679 non-null  object 
 5   District Area            660679 non-null  object 
 6   Longitude                660653 non-null  float64
 7   Number_of_Casualties     660679 non-null  int64  
 8   Number_of_Vehicles       660679 non-null  int64  
 9   Road_Surface_Conditions  659953 non-null  object 
 10  Road_Type                656159 non-null  object 
 11  Urban_or_Rural_Area      660664 non-null  object 
 12  Weather_Conditions       646551 non-null  object 
 13  Vehicle_Type             660679 non-null  object 
dtypes: f

In [9]:
accident.describe()

Latitude      Longitude  Number_of_Casualties  Number_of_Vehicles
count  660654.000000  660653.000000         660679.000000       660679.000000
mean       52.553866      -1.431210              1.357040            1.831255
std         1.406922       1.383330              0.824847            0.715269
min        49.914430      -7.516225              1.000000            1.000000
25%        51.490690      -2.332291              1.000000            1.000000
50%        52.315641      -1.411667              1.000000            2.000000
75%        53.453452      -0.232869              1.000000            2.000000
max        60.757544       1.762010             68.000000           32.000000

In [10]:
accident.dtypes

Index                       object
Accident_Severity           object
Accident Date               object
Latitude                   float64
Light_Conditions            object
District Area               object
Longitude                  float64
Number_of_Casualties         int64
Number_of_Vehicles           int64
Road_Surface_Conditions     object
Road_Type                   object
Urban_or_Rural_Area         object
Weather_Conditions          object
Vehicle_Type                object
dtype: object

<h2>CONVERTING OBJECT TO DATETIME DATA TYPE</h2>

In [11]:
accident['Accident Date'] = pd.to_datetime(accident['Accident Date'] , dayfirst = True, errors = 'coerce')

In [12]:
accident.dtypes

Index                              object
Accident_Severity                  object
Accident Date              datetime64[ns]
Latitude                          float64
Light_Conditions                   object
District Area                      object
Longitude                         float64
Number_of_Casualties                int64
Number_of_Vehicles                  int64
Road_Surface_Conditions            object
Road_Type                          object
Urban_or_Rural_Area                object
Weather_Conditions                 object
Vehicle_Type                       object
dtype: object

In [13]:
accident.isnull().sum()

Index                          0
Accident_Severity              0
Accident Date                  0
Latitude                      25
Light_Conditions               0
District Area                  0
Longitude                     26
Number_of_Casualties           0
Number_of_Vehicles             0
Road_Surface_Conditions      726
Road_Type                   4520
Urban_or_Rural_Area           15
Weather_Conditions         14128
Vehicle_Type                   0
dtype: int64

In [14]:
accident['Latitude'] = accident['Latitude'].fillna(accident['Latitude'].mode()[0])
accident['Longitude'] = accident['Longitude'].fillna(accident['Longitude'].mode()[0])
accident['Road_Surface_Conditions'] = accident['Road_Surface_Conditions'].fillna('unknown')
accident['Road_Type'] = accident['Road_Type'].fillna(accident['Road_Type'].mode()[0])
accident['Urban_or_Rural_Area'] = accident['Urban_or_Rural_Area'].fillna(accident['Urban_or_Rural_Area'].mode()[0])
accident['Weather_Conditions'] = accident['Weather_Conditions'].fillna('unknown')

In [15]:
accident['Year'] = accident['Accident Date'].dt.year
accident['Month'] = accident['Accident Date'].dt.month
accident['Day'] = accident['Accident Date'].dt.day
accident['DayofWeek'] = accident['Accident Date'].dt.dayofweek

In [16]:
accident['Index'] = accident['Index'].astype('category')
accident['Light_Conditions'] = accident['Light_Conditions'].astype('category')
accident['Latitude'] = accident['Latitude'].astype('category')
accident['Longitude'] = accident['Longitude'].astype('category')
accident['District Area'] = accident['District Area'].astype('category')
accident['Road_Type'] = accident['Road_Type'].astype('category')
accident['Urban_or_Rural_Area'] = accident['Urban_or_Rural_Area'].astype('category')
accident['Year'] = accident['Year'].astype('category')
accident['Month'] = accident['Month'].astype('category')
accident['Day'] = accident['Day'].astype('category')
accident['DayofWeek'] = accident['DayofWeek'].astype('category')

In [17]:
accident.dtypes

Index                            category
Accident_Severity                  object
Accident Date              datetime64[ns]
Latitude                         category
Light_Conditions                 category
District Area                    category
Longitude                        category
Number_of_Casualties                int64
Number_of_Vehicles                  int64
Road_Surface_Conditions            object
Road_Type                        category
Urban_or_Rural_Area              category
Weather_Conditions                 object
Vehicle_Type                       object
Year                             category
Month                            category
Day                              category
DayofWeek                        category
dtype: object

In [18]:
accident.isnull().sum()

Index                      0
Accident_Severity          0
Accident Date              0
Latitude                   0
Light_Conditions           0
District Area              0
Longitude                  0
Number_of_Casualties       0
Number_of_Vehicles         0
Road_Surface_Conditions    0
Road_Type                  0
Urban_or_Rural_Area        0
Weather_Conditions         0
Vehicle_Type               0
Year                       0
Month                      0
Day                        0
DayofWeek                  0
dtype: int64

In [19]:
df_2019 = accident[accident['Year'] == 2019]
df_2020 = accident[accident['Year'] == 2020]
df_2021 = accident[accident['Year'] == 2021]
df_2022 = accident[accident['Year'] == 2022]


<h2>INSIGHT #1</h2>

<h3>What is the total number of accidents in 2022</h3>

In [20]:
total_accidents = len(df_2022)

In [21]:
total_accidents

144419

<h3>The total number of accidents occured in 2022 is 144,419.</h3>

<h2>INSIGHT #2</h2>

<h3>How does the total number of accidents vary across months in 2022?</h3>

In [22]:
monthly_accidents = df_2022['Month'].value_counts().sort_index()

In [23]:
monthly_accidents

Month
1      9967
2     10935
3     12341
4     11510
5     12372
6     12812
7     12653
8     12088
9     12960
10    13534
11    13622
12     9625
Name: count, dtype: int64

<h3>Accidents peak in November , accounting for 9.43% of the total annual accidents, while December sees the lowest number of accidents (6.67% ), likely due to reduced travel during the holiday season.</h3>

<h2>INSIGHT #3</h2>

<h3>Which months have the highest number of casualties, and how do they compare across the year?</h3>

In [24]:
casualties_by_month = df_2022.groupby('Month')['Number_of_Casualties'].sum()

In [25]:
casualties_by_month

Month
1     13163
2     14804
3     16575
4     15767
5     16775
6     17230
7     17201
8     16796
9     17500
10    18287
11    18439
12    13200
Name: Number_of_Casualties, dtype: int64

<h3>The most dangerous months are November (18,439 casualties) and October (18,287 casualties) , accounting for the highest casualty counts, while February (14,804 casualties) and January (13,163 casualties) see relatively fewer casualties, likely due to reduced travel or activity during colder months.</h3>

<h2>INSIGHT #4</h2>

<h3>How do weather conditions influence the frequency of accidents?</h3>

In [26]:
weather_accidents = df_2022['Weather_Conditions'].value_counts()

In [27]:
weather_accidents

Weather_Conditions
Fine no high winds       115970
Raining no high winds     15195
Other                      4300
Snowing no high winds      2923
unknown                    2654
Raining + high winds       1140
Fine + high winds          1051
Fog or mist                 902
Snowing + high winds        284
Name: count, dtype: int64

<h3>The majority of accidents occur under "Fine no high winds" conditions, accounting for 63.68% of all accidents, while adverse weather conditions like snow or high winds contribute significantly less, with "Snowing + high winds" being the rarest at just 0.16% , highlighting that most accidents happen in clear or mild weather.</h3>

<h2>INSIGHT #5</h2>

<h3>How does the number of fatal accidents vary by road type?</h3>

In [28]:
fatal_accidents = df_2022[df_2022['Accident_Severity'] == 'Fatal']
fatal_road_accidents = fatal_accidents['Road_Type'].value_counts()

In [29]:
fatal_road_accidents

Road_Type
Single carriageway    1139
Dual carriageway       361
Roundabout              28
One way street          12
Slip road                9
Name: count, dtype: int64

<h3>Fatal accidents are most common on single carriageways , accounting for the majority of cases (1,139 incidents ), while other road types like slip roads (9 incidents ) and one-way streets (12 incidents ) see significantly fewer fatalities, highlighting the higher risk associated with single carriageways.</h3>

<h2>INSIGHT #6</h2>

<h3>How are accidents distributed across different severity levels?</h3>

In [30]:
severity_counts = df_2022['Accident_Severity'].value_counts()

In [31]:
severity_counts

Accident_Severity
Slight     124093
Serious     18777
Fatal        1549
Name: count, dtype: int64

<h3>The vast majority of accidents are classified as slight, accounting for 86.13% of all incidents, while serious accidents make up 12.98%, and fatal accidents represent only 1.07%, highlighting the predominance of less severe outcomes in accident data.</h3>

<h2>INSIGHT #7</h2>

<h3>How does the number of multi-vehicle accidents vary by month, and what percentage of total multi-vehicle accidents do they represent?</h3>

In [32]:
multi_vehicle_accidents = df_2022[df_2022['Number_of_Vehicles'] > 1]
multi_vehicle_monthly_accidents = multi_vehicle_accidents['Month'].value_counts().sort_index()

In [33]:
multi_vehicle_monthly_accidents

Month
1     6544
2     7378
3     8615
4     8268
5     8770
6     9240
7     9088
8     8721
9     9271
10    9628
11    9473
12    6370
Name: count, dtype: int64

<h3>Multi-vehicle accidents peak in October (9,628 incidents, 8.75%) and are lowest in December (6,370 incidents, 5.79%), with an overall trend showing higher frequencies during late summer and autumn months, likely due to increased traffic and travel activity.</h3>

<h2>INSIGHT #8</h2>

<h3>How do road surface conditions influence the frequency of accidents?</h3>

In [34]:
road_surface_counts = df_2022['Road_Surface_Conditions'].value_counts()

In [35]:
road_surface_counts

Road_Surface_Conditions
Dry                     98684
Wet or damp             35694
Frost or ice             6813
Snow                     2949
unknown                   165
Flood over 3cm. deep      114
Name: count, dtype: int64

<h3>The majority of accidents occur on dry roads , accounting for 69.74% of all incidents, while wet or damp conditions contribute 25.20% , and more hazardous conditions like frost, ice, or snow account for less than 5% combined, highlighting that most accidents happen under relatively normal road conditions.</h3>

<h2>INSIGHT #9</h2>

<h3>How does the total number of accidents compare across the years 2019 to 2022?</h3>

In [36]:
total_accidents_by_year = accident.groupby('Year').size()

In [37]:
total_accidents_by_year

Year
2019    182115
2020    170591
2021    163554
2022    144419
dtype: int64

<h3>The total number of accidents decreased consistently from 2019 (182,115) to 2022 (144,419), with a 20.7% overall reduction, reflecting a significant downward trend over the four years.</h3>

<h2>INSIGHT #10</h2>

<h3>How does the total number of casualties compare across the years 2019 to 2022?</h3>

In [38]:
total_casualties_by_year = accident.groupby('Year')['Number_of_Casualties'].sum()

In [39]:
total_casualties_by_year

Year
2019    247780
2020    230905
2021    222146
2022    195737
Name: Number_of_Casualties, dtype: int64

<h3>The total number of casualties decreased steadily from 2019 (247,780) to 2022 (195,737), reflecting a consistent downward trend over the four years.</h3>

<h2>INSIGHT #11</h2>

<h3>How does the average number of casualties per accident compare across the years 2019 to 2022?</h3>

In [40]:
avg_casualties_by_year = accident.groupby('Year')['Number_of_Casualties'].mean()

In [41]:
avg_casualties_by_year

Year
2019    1.360569
2020    1.353559
2021    1.358243
2022    1.355341
Name: Number_of_Casualties, dtype: float64

<h3>The average number of casualties per accident remained relatively stable between 1.35 and 1.36 across all four years, indicating no significant change in the severity of accidents over time.</h3>

<h2>INSIGHT #12</h2>

<h3>How does the number of fatal accidents compare across the years 2019 to 2022?</h3>

In [42]:
fatal_accidents = accident[accident['Accident_Severity'] == 'Fatal']
fatal_accidents_by_year = fatal_accidents.groupby('Year').size()

In [43]:
fatal_accidents_by_year

Year
2019    2714
2020    2341
2021    2057
2022    1549
dtype: int64

<h3>The number of fatal accidents decreased consistently from 2019 (2,714) to 2022 (1,549), reflecting a significant downward trend and indicating a reduction in the most severe accident outcomes over the four years.</h3>

<h2>INSIGHT #13</h2>

<h3>How does the distribution of accidents between urban and rural areas compare across the years 2019 to 2022?</h3>

In [44]:
urban_rural_accidents_by_year = accident.groupby(['Year', 'Urban_or_Rural_Area']).size().unstack()

In [45]:
urban_rural_accidents_by_year

Urban_or_Rural_Area  Rural  Unallocated   Urban
Year                                           
2019                 67558            5  114552
2020                 61991            6  108594
2021                 59016            0  104538
2022                 50425            0   93994

<h3>Accidents in urban areas consistently outnumber those in rural areas, with urban accidents decreasing from 114,552 in 2019 to 93,994 in 2022, and rural accidents also declining from 67,558 in 2019 to 50,425 in 2022, reflecting an overall reduction in accidents across both settings.</h3>

<h2>INSIGHT #14</h2>

<h3>How does the total number of casualties by road type compare across the years 2019 to 2022?</h3>

In [46]:
casualties_by_road_type = accident.groupby(['Year', 'Road_Type'])['Number_of_Casualties'].sum().unstack()

In [47]:
casualties_by_road_type

Road_Type  Dual carriageway  One way street  Roundabout  Single carriageway  \
Year                                                                          
2019                  41549            4570       14798              184053   
2020                  37960            4213       14459              171738   
2021                  35456            3890       14145              166197   
2022                  31912            3499       12683              145422   

Road_Type  Slip road  
Year                  
2019            2810  
2020            2535  
2021            2458  
2022            2221

<h3>Casualties on single carriageways consistently accounted for the largest share, decreasing from 184,053 in 2019 to 145,422 in 2022, while other road types like dual carriageways, one-way streets, roundabouts, and slip roads also showed a general downward trend, reflecting an overall reduction in casualties across all road types.</h3>

<h2>INSIGHT #15</h2>

<h3>How does the number of accidents under different weather conditions compare across the years 2019 to 2022?</h3>

In [48]:
weather_accidents_by_year = accident.groupby(['Year', 'Weather_Conditions']).size().unstack()

In [49]:
weather_accidents_by_year

Weather_Conditions  Fine + high winds  Fine no high winds  Fog or mist  Other  \
Year                                                                            
2019                             2545              143876          857   4083   
2020                             2861              132513          981   4265   
2021                             2097              128526          788   4502   
2022                             1051              115970          902   4300   

Weather_Conditions  Raining + high winds  Raining no high winds  \
Year                                                              
2019                                2959                  22835   
2020                                3130                  21984   
2021                                2386                  19682   
2022                                1140                  15195   

Weather_Conditions  Snowing + high winds  Snowing no high winds  unknown  
Year                                                                      
2019                                 146                    680     4134  
2020                                 201                    719     3937  
2021                                 254                   1916     3403  
2022                                 284                   2923     2654

<h3>Accidents occurring in "Fine no high winds" conditions consistently accounted for the majority, with a notable decline from 143,876 in 2019 to 115,970 in 2022. Accidents in adverse weather conditions like snow showed variability, while overall accident numbers decreased across all weather categories, reflecting both improved safety and potential changes in driving patterns over the years.</h3>

<h2>INSIGHT #16</h2>

<h3>How does the number of multi-vehicle accidents compare across the years 2019 to 2022?</h3>

In [50]:
multi_vehicle_accidents = accident[accident['Number_of_Vehicles'] > 1]
multi_vehicle_by_year = multi_vehicle_accidents.groupby('Year').size()

In [51]:
multi_vehicle_by_year

Year
2019    127182
2020    118086
2021    113258
2022    101366
dtype: int64

<h3>The number of multi-vehicle accidents decreased consistently from 127,182 in 2019 to 101,366 in 2022, reflecting a steady downward trend and indicating a reduction in such incidents over the four years.</h3>

<h2>INSIGHT #17</h2>

<h3>How does the number of accidents vary by month across the years 2019 to 2022?</h3>

In [52]:
monthly_accidents_by_year = accident.groupby(['Year', 'Month']).size().unstack()

In [53]:
monthly_accidents_by_year

Month      1      2      3      4      5      6      7      8      9     10  \
Year                                                                          
2019   15355  13253  15049  14125  15833  15528  15862  15044  15271  15528   
2020   14133  14353  13494  13394  14336  14205  14630  13366  14432  15684   
2021   13417  10950  13202  12715  13811  13936  14300  13415  13792  14834   
2022    9967  10935  12341  11510  12372  12812  12653  12088  12960  13534   

Month     11     12  
Year                 
2019   16559  14708  
2020   14770  13794  
2021   15473  13709  
2022   13622   9625

<h3>Accidents tend to peak in November and December, with a general decline in total monthly accidents over the years. For instance, December saw 14,708 accidents in 2019 but only 9,625 in 2022, reflecting an overall downward trend while maintaining seasonal patterns.</h3>

<h2>INSIGHT #18</h2>

<h3>How does the number of fatal accidents in urban areas compare across the years 2019 to 2022?</h3>

In [54]:
fatal_urban_accidents = accident[(accident['Accident_Severity'] == 'Fatal') & (accident['Urban_or_Rural_Area'] == 'Urban')]
fatal_urban_by_year = fatal_urban_accidents.groupby('Year').size()

In [55]:
fatal_urban_by_year

Year
2019    941
2020    846
2021    734
2022    539
dtype: int64

<h3>Fatal accidents in urban areas decreased consistently from 941 in 2019 to 539 in 2022, reflecting a significant downward trend and indicating improved safety measures or reduced exposure to high-risk scenarios in urban settings over the years.</h3>

<h2>INSIGHT #19</h2>

<h3>How does the number of multi-vehicle accidents on single carriageways compare across the years 2019 to 2022?</h3>

In [56]:
multi_vehicle_single_accidents = accident[(accident['Number_of_Vehicles'] > 1) & (accident['Road_Type'] == 'Single carriageway')]
multi_vehicle_single_by_year = multi_vehicle_single_accidents.groupby('Year').size()

In [57]:
multi_vehicle_single_by_year

Year
2019    92972
2020    86278
2021    83240
2022    74101
dtype: int64

<h3>The number of multi-vehicle accidents on single carriageways decreased steadily from 92,972 in 2019 to 74,101 in 2022, indicating a consistent downward trend and suggesting improvements in road safety or changes in traffic patterns on these road types over the years.</h3>

<h2>INSIGHT #20</h2>

<h3>How does the number of daylight accidents in rural areas compare across the years 2019 to 2022?</h3>

In [58]:
rural_daylight_accidents = accident[(accident['Urban_or_Rural_Area'] == 'Rural') & (accident['Light_Conditions'] == 'Daylight')]
rural_daylight_by_year = rural_daylight_accidents.groupby('Year').size()

In [59]:
rural_daylight_by_year

Year
2019    49359
2020    45212
2021    43133
2022    37646
dtype: int64

<h3>Daylight accidents in rural areas decreased consistently from 49,359 in 2019 to 37,646 in 2022, reflecting a steady downward trend and potentially indicating improvements in road safety or reduced traffic volumes in rural regions over the years.</h3>